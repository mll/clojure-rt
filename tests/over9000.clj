(defn over9000
  [x]
  (if (< 9000 x)
    x
    (over9000 (+ x x))))

(over9000 1)

(defn over9000-recur
  [x]
  (if (< 9000 x)
    x
    (recur (+ x x))))

(over9000-recur 1)

;; CHECK: define i64 @fn_2_J_J
;; CHECK-NOT: load atomic ptr
;; CHECK: }

(defn over9000-recur-1
  ([x]
   (if (< 9000 x)
     x
     (recur (+ x x))))
  ([x y] 123))

(over9000-recur-1 1)
(over9000-recur-1 1 2)

(defn over9000-recur-2
  ([x y] 123)
  ([x]
   (if (< 9000 x)
     x
     (recur (+ x x)))))

(over9000-recur-2 1)
(over9000-recur-2 1 2)
