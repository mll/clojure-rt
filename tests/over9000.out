Compiling to AST...
Success!!
Executing...
Ref counters: 0 2 0 1 1 0 1 0 0 1 0 2 0 0 0 0 0 
Computing: (def over9000 (clojure.core/fn ([x] (if (< 9000 x) x (over9000 (+ x x))))))
Printing...
Result: nil
Ref counters: 0 2 0 1 1 0 1 0 0 1 0 2 0 1 0 0 0 
Ref counters: 0 2 0 1 1 0 1 0 0 1 0 2 0 1 0 0 0 
Computing: (over9000 1)
Printing...
Result: 16384
Ref counters: 0 2 0 1 1 0 1 0 0 1 0 2 0 1 0 0 0 
Ref counters: 0 2 0 1 1 0 1 0 0 1 0 2 0 1 0 0 0 
Computing: (def over9000-recur (clojure.core/fn ([x] (if (< 9000 x) x (recur (+ x x))))))
Printing...
Result: nil
Ref counters: 0 2 0 1 1 0 1 0 0 1 0 2 0 2 0 0 0 
Ref counters: 0 2 0 1 1 0 1 0 0 1 0 2 0 2 0 0 0 
Computing: (over9000-recur 1)
Printing...
Result: 16384
Ref counters: 0 2 0 1 1 0 1 0 0 1 0 2 0 2 0 0 0 
